ogr/ogrsf_frmts/osm/ogrosmdatasource.cpp

```diff
@@ -205,13 +205,14 @@ static CPLString GetInterestLayersForDSName(const CPLString &osDSName)
 {
     CPLMutexHolder oMutexHolder(&hMutex);
     GIntBig nPID = CPLGetPID();
-    for (int i = 0; i < (int)oListDSToBeOpened.size(); i++)
+    for (auto oIter = oListDSToBeOpened.begin();
```

Comment:
can you use a Range-based for loop here?  (since you are modernizing the code)

Reply:
actually no, because in the loop, we user the iterator as the argument of the std::vector::erase() method